File: src/ExpoPrinterModule.ts
import { NativeModule, requireNativeModule } from "expo";

import { XpTt426bLabel } from "./ExpoPrinter.types";

declare class ExpoPrinterModule extends NativeModule {
  getVersion(): string;

  hasBluetoothPermissions(): boolean;

  getRequiredPermissions(): string[];

  connect(address: string): Promise<string>;

  isConnected(): boolean;

  printTest(): void;

  disconnect(): boolean;

  printXpTt426bLabel(data: XpTt426bLabel): Promise<void>;
  
}

export default requireNativeModule<ExpoPrinterModule>("ExpoPrinter");

File: android/src/main/java/expo/modules/printer/ExpoPrinterModule.kt
package expo.modules.printer

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoPrinterModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoPrinter")

    Function("getVersion") {
      "1.0.0"
    }
  }
}